### ISSUE TITLE
Compatibility with other react-native modules

### ISSUE THREAD

**Issue by drorbiran** [wix/react-native-navigation#242]
>This issue was originally opened in [react-native-controllers](https://github.com/wix/react-native-controllers/issues/40).

One of the challenges migrating to react-native-controllers is that a lot of the other extensions stop working, because they assume the root controller never changes (and is the top one in the window hierarchy).

To fix I had to change every single dependency to lazily fetch the root controller, and call `presentedViewController` on it, like:

```obj-c
-(UIViewController *)getRootController
{
  id<UIApplicationDelegate> appDelegate = [UIApplication sharedApplication].delegate;
  return appDelegate.window.rootViewController.presentedViewController;
}
```

Is there any way we can fix this more broadly from react-native-controllers itself? Or any suggestions of upstream changes that would allow this to coexist with other extensions?

Thanks!

**Comment by mrpatiwi** [comment]
+1
I can't make it work with external native-modules like https://github.com/airbnb/react-native-maps.

I hope there is a way to make `react-native-navigation` more friendly with the rest of the modules out there.

**Comment by akilbekov** [comment]
@mrpatiwi did you manage to fix it?

**Comment by guyca** [comment]
What exactly doesn't work for you guys?

**Comment by hammadj** [comment]
@guyca 

For example, in my app, I am using the Stripe SDK for iOS: https://stripe.com/docs/mobile/ios/custom

To use this I am creating a [native bridge](https://facebook.github.io/react-native/docs/native-modules-ios.html) in Objective C. To display the payment form, I have this method in Objective C that I call from React Native: 
```obj-c
- (void)beginCustomPaymentWithAmount:(NSString *)amount {
  PaymentViewController *paymentViewController = [[PaymentViewController alloc] initWithNibName:nil bundle:nil];
  paymentViewController.amount = amount;
  paymentViewController.delegate = self;
  UINavigationController *navController = [[UINavigationController alloc] initWithRootViewController:paymentViewController];
  [rootViewController presentViewController:navController animated:YES completion:nil];
}
```

The root view controller is defined in the following method used to initialize the Stripe SDK:
```obj-c
RCT_EXPORT_METHOD(initWithStripePublishableKey:(NSString *)stripeKey applePayMerchantId:(NSString *)merchantId) {
  _initialized = TRUE;
  applePayMerchantId = merchantId;
  [Stripe setDefaultPublishableKey:stripeKey];

  id<UIApplicationDelegate> appDelegate = [UIApplication sharedApplication].delegate;
  rootViewController = appDelegate.window.rootViewController.presentedViewController;
}
```

However, this doesn't work when using react-native-navigation. How can I get the appropriate view controller so I could present a new screen from the native side?

**Comment by hamidhadi** [comment]
I have the same issue with [https://github.com/airbnb/react-native-maps]( https://github.com/airbnb/react-native-maps)  on android.

**Comment by DavidKuennen** [comment]
Same issue. Some way to fix this? It's kind of a show stopper for me.

**Comment by DavidKuennen** [comment]
Ok, somehow it's loading now. No idea why tho.

**Comment by hamidhadi** [comment]
@DavidKuennen If you can see the Google logo, it seems you have problem with Google Maps Api Key.

**Comment by DavidKuennen** [comment]
@hamidhadi Yes, either that or there was some problem with my internet connection. Thank you.
